price-aggregator/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+#![no_std]
+#![feature(destructuring_assignment)]
+
+elrond_wasm::imports!();
+pub mod median;
+mod price_aggregator_data;
+
+use elrond_wasm::require;
+use price_aggregator_data::{OracleStatus, PriceFeed, TokenPair};
+
+#[elrond_wasm_derive::contract]
+pub trait PriceAggregator {
+    #[init]
+    fn init(
+        &self,
+        payment_token: TokenIdentifier,
+        oracles: Vec<Address>,
+        submission_count: u32,
+        decimals: u8,
+        query_payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.payment_token().set(&payment_token);
+        self.query_payment_amount().set(&query_payment_amount);
+        self.submission_count().set(&submission_count);
```

Comment:
we might need an endpoint so the owner can change the threshold for feed aggregation

Reply:
Added `setSubmissionCount` endpoint, with argument `submission_count: u32` callable only by the owner.